### PR TITLE
Fix Cargo Changes

### PR DESCRIPTION
**Pull request by FatemeYaghooblou**
@mohammadKarimi for review

### CODE REVIEW

File: Examples/E3/Cargo.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        IShipping CreateShipping();
+    }
+
+    public class AirFactory : IShippingFactory
+    {
+        private static Air _air;
+
+        public IShipping CreateShipping()
+        {
+            if (_air == null)
```
Comment (mohammadKarimi) [inline Examples/E3/Cargo.cs:73]: consider it is not thread-safe . but totally for example it is enough.

File: Examples/E3/Cargo.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            {
+                _air = new Air();
+            }
+
+            return _air;
+        }
+    }
+
+    public class ShipFactory : IShippingFactory
```
Comment (mohammadKarimi) [inline Examples/E3/Cargo.cs:82]: perfect.

File: Examples/E3/Cargo.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        }
+
+    }
+    public class TrainFactory : IShippingFactory
+    {
+
+        public IShipping CreateShipping()
+        {
+            Train train = new Train();
+            Call();
```
Comment (mohammadKarimi) [inline Examples/E3/Cargo.cs:106]: call in very general naming, 
Initialize() is much more better.

**Review by mohammadKarimi — COMMENTED** [review]
perfect one.

perfect,

now I want you to make E3 more complex,

E3.1
Imagine it is a library and we make it as a package. so users or customers, can't change your code, but they want to add a new behavior like Truck. so, how you can solve it.

**Comment by FatemeYaghooblou** [comment]
@mohammadKarimi  I've refactored the Cargo class based on the E3.1 practice

File: Examples/E3/Cargo.cs
Comment (FatemeYaghooblou) [inline Examples/E3/Cargo.cs:1]: @mohammadKarimi to remind you for review.

**Pull request closed without merging**